core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -236,37 +252,22 @@ public static LoadTableResponse loadTable(Catalog catalog, TableIdentifier ident
     throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
   }
 
+  public static void renameTable(Catalog catalog, RenameTableRequest request) {
+    catalog.renameTable(request.source(), request.destination());
+  }
+
   public static LoadTableResponse updateTable(Catalog catalog, TableIdentifier ident, UpdateTableRequest request) {
     TableMetadata finalMetadata;
     if (isCreate(request)) {
-      // this is a hacky way to get TableOperations for an uncommitted table
-      Transaction transaction = catalog.buildTable(ident, EMPTY_SCHEMA).createOrReplaceTransaction();
-      if (transaction instanceof BaseTransaction) {
-        BaseTransaction baseTransaction = (BaseTransaction) transaction;
-        finalMetadata = create(baseTransaction.underlyingOps(), baseTransaction.startMetadata(), request);
-      } else {
-        throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTransaction");
-      }
-
+      finalMetadata = commitCreate(catalog, ident, request);
     } else {
-      Table table = catalog.loadTable(ident);
-      if (table instanceof BaseTable) {
-        TableOperations ops = ((BaseTable) table).operations();
-        finalMetadata = commit(ops, request);
-      } else {
-        throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
-      }
+      finalMetadata = commitUpdate(catalog, ident, request);
     }
-
```

Comment:
Nit: we like to keep a newline separating control flow blocks from the following statements.

Reply:
This is done.